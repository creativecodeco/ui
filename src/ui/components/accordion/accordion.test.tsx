import { render } from '@testing-library/react';

import type { AccordionOption, AccordionType } from '@/types';

import Accordion from './accordion.component';

const baseProps: AccordionType = {
  name: 'test',
  options: Array.from({ length: 4 }).map<AccordionOption>((_, index) => ({
    header: <>Header {index}</>,
    body: <p>Body</p>
  }))
};

describe('<Accordion />', () => {
  it('snapshot', () => {
    const { container } = render(<Accordion {...baseProps} />);

    expect(container).toMatchSnapshot();
  });

  it('snapshot all props', () => {
    const { container } = render(
      <Accordion {...baseProps} iconType='plus' multiple join />
    );

    expect(container).toMatchSnapshot();
  });

  it('options - single', () => {
    const { getAllByRole } = render(<Accordion {...baseProps} />);

    expect(getAllByRole('radio').length).toEqual(4);
  });

  it('options - multiple', () => {
    const { getAllByRole } = render(<Accordion {...baseProps} multiple />);

    expect(getAllByRole('checkbox').length).toEqual(4);
  });

  it('not join', () => {
    const { container } = render(<Accordion {...baseProps} />);

    expect(container.querySelector('div')?.className).toEqual(
      'flex flex-wrap gap-2'
    );
  });

  it('join', () => {
    const { container } = render(<Accordion {...baseProps} join />);

    expect(container.querySelector('div')?.className).toEqual(
      'flex flex-wrap join join-vertical w-full'
    );
  });
});
