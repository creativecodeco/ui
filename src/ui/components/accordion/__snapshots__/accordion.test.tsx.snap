// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> snapshot 1`] = `
<div>
  <div
    class="flex flex-wrap gap-2"
  >
    <div
      class="collapse bg-base-200 accordion-type-arrow"
    >
      <input
        name="test"
        type="radio"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        0
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
    <div
      class="collapse bg-base-200 accordion-type-arrow"
    >
      <input
        name="test"
        type="radio"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        1
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
    <div
      class="collapse bg-base-200 accordion-type-arrow"
    >
      <input
        name="test"
        type="radio"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        2
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
    <div
      class="collapse bg-base-200 accordion-type-arrow"
    >
      <input
        name="test"
        type="radio"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        3
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> snapshot all props 1`] = `
<div>
  <div
    class="flex flex-wrap join join-vertical w-full"
  >
    <div
      class="collapse bg-base-200 accordion-type-plus join-item border border-base-300"
    >
      <input
        name="test"
        type="checkbox"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        0
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
    <div
      class="collapse bg-base-200 accordion-type-plus join-item border border-base-300"
    >
      <input
        name="test"
        type="checkbox"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        1
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
    <div
      class="collapse bg-base-200 accordion-type-plus join-item border border-base-300"
    >
      <input
        name="test"
        type="checkbox"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        2
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
    <div
      class="collapse bg-base-200 accordion-type-plus join-item border border-base-300"
    >
      <input
        name="test"
        type="checkbox"
      />
      <div
        class="collapse-title text-xl font-medium"
      >
        Header 
        3
      </div>
      <div
        class="collapse-content"
      >
        <p>
          Body
        </p>
      </div>
    </div>
  </div>
</div>
`;
